modules/fedimint-ln-client/src/lib.rs

```diff
@@ -404,30 +486,30 @@ impl ClientModuleGen for LightningClientGen {
         &self,
         cfg: Self::Config,
         _db: Database,
-        _module_root_secret: DerivableSecret,
+        module_root_secret: DerivableSecret,
         notifier: ModuleNotifier<DynGlobalClientContext, <Self::Module as ClientModule>::States>,
         _api: DynGlobalApi,
-        _module_api: DynModuleApi,
+        module_api: DynModuleApi,
     ) -> anyhow::Result<Self::Module> {
         Ok(LightningClientModule {
             cfg,
             notifier,
+            redeem_key: module_root_secret
+                .child_key(ChildId(0))
+                .to_secp_key(&Secp256k1::new()),
             secp: secp256k1_zkp::Secp256k1::new(),
```

Comment:
It's kind of weird that we instantiate two secp contexts here? Can we use the same one? I forget why we need the zkp context here, maybe @elsirion knows

Reply:
shared it. lmk if there's some bad in sharing